nni/retiarii/converter/graph_gen.py

```diff
@@ -597,10 +579,15 @@ def convert_module(self, script_module, module, module_name, ir_model):
             pass  # do nothing
         elif original_type_name == OpTypeName.LayerChoice:
             graph = Graph(ir_model, -100, module_name, _internal=True)  # graph_id is not used now
-            candidate_name_list = [f'layerchoice_{module.label}_{cand_name}' for cand_name in module.names]
-            for cand_name, cand in zip(candidate_name_list, module):
+            candidate_name_list = []
+            for cand_name in module.names:
+                cand = module[cand_name]
+                script_cand = script_module._modules[cand_name]
+                cand_name = build_cand_name(cand_name, module.label)
+                candidate_name_list.append(cand_name)
                 cand_type = '__torch__.' + get_importable_name(cand.__class__)
-                graph.add_node(cand_name, cand_type, get_init_parameters_or_fail(cand))
+                graph.add_node(cand_name, cand_type, get_init_parameters_or_fail(cand, silently=True))
+                self._convert_module(script_cand, cand, cand_name, ir_model)
```

Comment:
I think choices in LayerChoice should stop parsing. Please test with examples in https://github.com/microsoft/nni/tree/master/test/retiarii_test/darts to see if that works.

Reply:
But nn-meter needs the subgraph of layerchoice's candidates.

If there are really some errors, users just need to wrap the candidate with `serialize` to stop parsing.

Reply:
Actually `serialize` is required under the former implementation. There is no difference if `serialize` is provided. What I implement is that when the candidate is not wrapped with `serialize`, I will parse it recursively.

Reply:
Make sense to me. If you have already tested the example, I'm okay with this change.

Reply:
suggest to add ut for non-serialize case.